settings.php

```diff
@@ -59,873 +59,95 @@
-		}
-		else
-		{
-			$piHoleLogging = false;
-		}
-	} else {
-		$piHoleLogging = true;
-	}
-?>
-		<div class="box box-primary">
-			<div class="box-header with-border">
-				<h3 class="box-title">Query Logging<?php if($piHoleLogging) { ?> (size of log <?php echo formatSizeUnits(filesize("/var/log/pihole.log")); ?>)<?php } ?></h3>
-			</div>
-			<div class="box-body">
-				<p>Current status:
-				<?php if($piHoleLogging) { ?>
-					Enabled (recommended)
-				<?php }else{ ?>
-					Disabled
-				<?php } ?></p>
-				<?php if($piHoleLogging) { ?>
-					<p>Note that disabling will render graphs on the web user interface useless</p>
-				<?php } ?>
-			</div>
-			<div class="box-footer">
-				<form role="form" method="post">
-				<button type="button" class="btn btn-default confirm-flushlogs">Flush logs</button>
-				<input type="hidden" name="field" value="Logging">
-				<input type="hidden" name="token" value="<?php echo $token ?>">
-				<?php if($piHoleLogging) { ?>
-					<input type="hidden" name="action" value="Disable">
-					<button type="submit" class="btn btn-primary pull-right">Disable query logging</button>
-				<?php } else { ?>
-					<input type="hidden" name="action" value="Enable">
-					<button type="submit" class="btn btn-primary pull-right">Enable query logging</button>
-				<?php } ?>
-				</form>
-			</div>
-		</div>
-<?php
-	// Excluded domains in API Query Log call
-	if(isset($setupVars["API_EXCLUDE_DOMAINS"]))
-	{
-		$excludedDomains = explode(",", $setupVars["API_EXCLUDE_DOMAINS"]);
-	} else {
-		$excludedDomains = [];
-	}
-
-	// Exluded clients in API Query Log call
-	if(isset($setupVars["API_EXCLUDE_CLIENTS"]))
-	{
-		$excludedClients = explode(",", $setupVars["API_EXCLUDE_CLIENTS"]);
-	} else {
-		$excludedClients = [];
-	}
-
-	// Exluded clients
-	if(isset($setupVars["API_QUERY_LOG_SHOW"]))
-	{
-		$queryLog = $setupVars["API_QUERY_LOG_SHOW"];
-	} else {
-		$queryLog = "all";
-	}
-
-	// Privacy Mode
-	if(isset($setupVars["API_PRIVACY_MODE"]))
-	{
-		$privacyMode = $setupVars["API_PRIVACY_MODE"];
-	} else {
-		$privacyMode = false;
-	}
-
+    $includeWidget = "";
+
+    $settingsWidget = $_GET["w"];
+    
+    $settingsWidgets = array(
+        "networking"    => "scripts/pi-hole/php/settings_networking.php",
+        "dhcp"          => "scripts/pi-hole/php/settings_dhcp.php",
+        "upstream_dns"  => "scripts/pi-hole/php/settings_upstream_dns.php",
+        "querylogging"  => "scripts/pi-hole/php/settings_querylogging.php",
+        "blocklist"     => "scripts/pi-hole/php/settings_blocklist.php",
+        "api"           => "scripts/pi-hole/php/settings_api.php",
+        "webui"         => "scripts/pi-hole/php/settings_webui.php",
+        "sysadmin"      => "scripts/pi-hole/php/settings_sysadmin.php",
+        "ftl"           => "scripts/pi-hole/php/settings_ftl.php",
+        "teleporter"    => "scripts/pi-hole/php/settings_teleporter.php",
+    );
+    
+    $includeWidget = $settingsWidgets[$settingsWidget];
```

Comment:
Yes, just like that. We are all free timers and when we work on individual PRs, then we often don't find the time to review others. Since your PR introduces many changes, I simply didn't get around to thoroughly look through your proposed code - hence the delay. There are other projects where PRs are mostly only checked for functionality, but we require a full code review for security reasons. Unfortunately, this can lead to huge delays, but we usually get there in the end